matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/room/timeline/TimelineEventDecryptor.kt

```diff
@@ -104,12 +105,16 @@ internal class TimelineEventDecryptor(
         try {
             val result = cryptoService.decryptEvent(event, timelineId)
             Timber.v("Successfully decrypted event ${eventId}")
-            eventEntity.setDecryptionResult(result)
+            realm.executeTransaction {
+                eventEntity.setDecryptionResult(result)
```

Comment:
How this code was working before and how can we avoid doing this mistake again?

Reply:
The code is working, but we were doing work in the transaction block, it can blocks other transactions for no reason. We just have to be vigilant I guess.